sympy/matrices/expressions/matexpr.py

```diff
@@ -127,12 +127,20 @@ def __pow__(self, other):
             raise ShapeError("Power of non-square matrix %s" % self)
         elif self.is_Identity:
             return self
-        elif other is S.NegativeOne:
+        elif other is S.NegativeOne and not isinstance(self,MatPow):
             return Inverse(self)
         elif other is S.Zero:
             return Identity(self.rows)
         elif other is S.One:
             return self
+
+        if(isinstance(self,MatPow)):
+            new_exp = sympify(self.exp * other)
+            if(isinstance(new_exp, Integer)):
```

Comment:
Do not put parentheses around the if. Just use `if isinstance(new_exp, Integer)`.